src/QsCompiler/SyntaxProcessor/SymbolOccurrence.fs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+namespace Microsoft.Quantum.QsCompiler.SyntaxProcessing
+
+open System
+open System.Collections.Immutable
+open System.Diagnostics.CodeAnalysis
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+
+type SymbolOccurrence =
+    | Declaration of QsSymbol
+    | UsedType of QsType
+    | UsedVariable of QsSymbol
+    | UsedLiteral of QsExpression
+
+module SymbolOccurrence =
+    let rec flattenSymbol symbol =
+        match symbol.Symbol with
+        | SymbolTuple items -> Seq.collect flattenSymbol items |> Seq.toList
+        | _ -> [ symbol ]
+
+    let symbolDeclarations = flattenSymbol >> List.map Declaration
```

Comment:
This used to not return anything for missing and invalid symbols (see also the comment on the type declaration).

Reply:
IMO, this is an improvement. For example, previous behavior is to show nothing when hovering over `_` in `let _ = ...`. Now it shows the hover text "Discarded symbol assignment."

Reply:
Yes, my main concern was that it may not work as expected in some of the places where it is used (e.g. type declarations came to mind). That there is no range and it doesn't show up makes sense, though.